### PR TITLE
Add lint and cover binaries to glide, glide and Makefile cleanup

### PR DESCRIPTION
**Pull request by peter-edge**
This is another step towards eventual docker integration.

- This adds the go binaries we `go get` in `make lintbins` and `make coverbins` to glide. The dependencies are a little too harsh (getting all bins for `lint` and `cover`) but this will be fixed when the docker integration PR comes.
- This also splits `imports` from `testImports` in glide, and alphabetizes the imports. This makes glide better for using yarpc as a dependency. I had to keep testify and mock in `imports` because non `_test.go` code depends on them.
- This also adds 'homepage' and 'license' to glide.
- Reduces matrix from 5 instances to 3 on Travis by adding `examples` to the test cases. This target does not take long and we're using extra parallel builds.
-  General Makefile cleanup.

### CODE REVIEW

**Comment by breerly** [comment]
I don't think you can rely on testImports actually. If I wasn't in bed I'd post a link to the Glide issue I filed.

File: Makefile [hunk 1]
```diff
@@ -111,70 +117,56 @@ PROTOC = $(BIN)/protoc
 BINS = $(DOCKER_COMPOSE) $(GLIDE) $(THRIFT) $(PROTOC) $(BIN)/thriftrw-plugin-yarpc $(BIN)/protoc-gen-yarpc-go
 
 $(DOCKER_COMPOSE):
-	mkdir -p $(BIN)
+	@mkdir -p $(BIN)
```
Comment (breerly) [inline Makefile]: Any reason to hide all this output?

Trying to manage what info is useful to look at?

**Comment by peter-edge** [comment]
testImports are just for tools right now, ie things we vendor so we can use their binaries, I think this should be ok? I've never had issues.

File: Makefile (hunk 1 above)
Comment (peter-edge) [inline Makefile]: Ya basically heh

**Pull request merged**